src/webapi/rendering_context.rs

```diff
@@ -27,7 +183,771 @@ impl RenderingContext for CanvasRenderingContext2d {
     }
 }
 
+impl CanvasGradient {
+
+    /// Adds a new stop, defined by an offset and a color, to the gradient. If the offset is 
+    /// not between 0 and 1, an INDEX_SIZE_ERR is raised, if the color can't be parsed as a 
+    /// CSS <color>, a SYNTAX_ERR is raised.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasGradient/addColorStop)
+    // https://html.spec.whatwg.org/#2dcontext:dom-canvasgradient-addcolorstop
+    pub fn add_color_stop(&self, offset: f64, color: &str) -> Result<(), SyntaxError> {
```

Comment:
This also can throw `IndexSizeError` and yet it's marked as only returning a `SyntaxError`.

You should probably do something like this:

```rust
error_enum_boilerplate! {
    AddColorStopError,
    SyntaxError, IndexSizeError
}
```

where `SyntaxError` should be imported from `webapi::dom_exception` and `IndexSizeError` should be added to `webapi::dom_exception`.